crates/bevy_hierarchy/src/child_builder.rs

```diff
@@ -279,39 +279,68 @@ pub struct ChildBuilder<'a> {
     push_children: PushChildren,
 }
 
-impl ChildBuilder<'_> {
+/// Trait for building children entities and adding them to a parent entity. This is used in
+/// implementations of [`BuildChildren`] as a bound on the [`Builder`](BuildChildren::Builder)
+/// associated type. The closure passed to [`BuildChildren::with_children`] accepts an
+/// implementation of `ChildBuild` so that children can be spawned via [`ChildBuild::spawn`].
+pub trait ChildBuild {
```

Comment:
Worth thinking about whether we want the user to be able to implement this trait on their own. The current API implies this and makes adding a new method a breaking change.
What do you think @alice-i-cecile?

Reply:
IMO we don't bother sealing this but we also just don't really care about adding new methods. This isn't long for the world: complexity isn't worth it here.